src/satosa/frontends/openid_connect.py

```diff
@@ -173,6 +173,18 @@ def register_endpoints(self, backend_names):
         :rtype: list[(str, ((satosa.context.Context, Any) -> satosa.response.Response, Any))]
         :raise ValueError: if more than one backend is configured
         """
+        # See https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
+        # Unfortunately since the issuer is always `base_url` for all OIDC frontend instances,
+        # the discovery endpoint will be the same for every instance.
+        # This means that only one frontend will be usable for autodiscovery.
```

Comment:
> OpenID Providers supporting Discovery MUST make a JSON document available at the path formed by concatenating the string `/.well-known/openid-configuration` to the Issuer.

The issuer is discovered through a WebFinger request for resources of `http://openid.net/specs/connect/1.0/issuer` relation. The response contains one or more `href` properties with the Issuer URL, which is allowed to contain a path.

What we define is that the frontend contains the frontend name as a path component and under that you can query the well-known documents.

With that in mind we can have multiple frontends each with its own discovery.

The problem is that atm, the `base_url` is used instead of `endpoint_baseurl`.

We can introduce a configuration option to select between the two behaviours, or (even better) introduce a configuration to set the discovery URL for a frontend.

---

At some point I would like to invert this logic; instead of a component defining paths of URLs internally that mapped to functionality (which the routing module has to match to based on some rules), there should be URLs as entrypoints mapped to functionality directly (as it happens within most web frameworks - flask, django, fastapi, etc).

Reply:
> The problem is that atm, the base_url is used instead of endpoint_baseurl.
> We can introduce a configuration option to select between the two behaviours, or (even better) introduce a configuration to set the discovery URL for a frontend.

Would you agree to add a `use_module_name_in_issuer` option (default False for backward compatibility, but the examples changed to True)?

A more subtle change but also harder to document alternative would be to make the assignment in [https://github.com/IdentityPython/SATOSA/blob/fc9374c9959ddd57d69ee69f9f683f96ddede964/src/satosa/frontends/openid_connect.py#L66](__init__.py:66)  optional, so that one could set the issuer manually under the `provider` array.

Is any of the two OK with you, or am I misunderstanding the problem?

Reply:
Yes, both are ok with me. As long as we provide a way to configure things to work as before, it is fine to introduce such changes.

Reply:
The updated patchset contains a commit, which does not perform the above assignment if the provider config has the issuer set. I did some research in git log, but I think it was just in this way forever.
I added a brief explanation to the example configuration, too.
Note that this could be a breaking change for those, who had a lurking "issuer" in their provider config, but since it's never been supported, I'm inclined to go this way rather than adding a new "fix-something-but-dont-break-old-config" type of configuration option.